### ISSUE TITLE
bearer token support

### ISSUE THREAD

**Issue by theoden8** [kevinhermawan/Ollamac#69]
Good afternoon,

Would it be possible to ask for bearer token support to authorize with protected self-hosted ollama service?

The implementation detail is that the HTTP/HTTPS request needs to contain the following header field:

```
authorization: Bearer $token
```

Kind regards

**Comment by kevinhermawan** [comment]
Thanks for the suggestion! Let’s make it happen.

**Comment by gromer** [comment]
Yeah, this would be nice. After poking around at the code, looks like an update to the underlying [OllamaKit library](https://github.com/kevinhermawan/OllamaKit) would be needed to add the additional Authorization header support before Ollamac could actually send the token. Most likely at least changes to:

1. [OllamaKit library](https://github.com/kevinhermawan/OllamaKit)
    1. [OllamaKit](https://github.com/kevinhermawan/OllamaKit/blob/main/Sources/OllamaKit/OllamaKit.swift) to add additional constructors to manage the additional param.
        1. Adding maybe `init(baseURL: URL, bearerToken: String)` and `init(bearerToken: String)`
    2. [OKRouter](https://github.com/kevinhermawan/OllamaKit/blob/main/Sources/OllamaKit/Utils/OKRouter.swift) to update the `asURLRequest` function to take in a `bearerToken` param to modify the request headers.
    3. All the callers of `asURLRequest` so include the bearer token, if set.

Then updates would be needed to Ollamac to provide the ability to set the bearer token and leverage it when set.

I haven't done much Swift, but may have time to give it a try.
